cvat/apps/engine/media_extractors.py

```diff
@@ -510,3 +553,190 @@ def _is_zip(path):
         'unique': True,
     }
 }
+
+
+class ValidateDimension:
+
+    def __init__(self, path=None):
+        self.dimension = "2d"
+        self.path = path
+        self.related_files = {}
+        self.image_files = {}
+        self.converted_files = []
+
+    @staticmethod
+    def validate_pcd(path):
+        """
+            Validate the point cloud file along with version check
+            On failure it is considered as PhotoCD format
+        """
+
+        pcd_version = ["0.7", "0.6", "0.5", "0.4", "0.3", "0.2", "0.1", ".7", ".6", ".5", ".4", ".3", ".2", ".1"]
+        with open(path, "rb") as file_read:
+            data = file_read.read(70)
+            for ver in pcd_version:
+                version = f"VERSION {ver}".encode()
+                if version in data:
+                    return True
+        return False
+
+    @staticmethod
+    def convert_bin_to_pcd(path, delete_source=True):
+        list_pcd = []
+        with open(path, "rb") as f:
+            size_float = 4
+            byte = f.read(size_float * 4)
+            while byte:
+                x, y, z, intensity = struct.unpack("ffff", byte)
+                list_pcd.append([x, y, z])
+                byte = f.read(size_float * 4)
+        np_pcd = np.asarray(list_pcd)
+        pcd = o3d.geometry.PointCloud()
+        pcd.points = o3d.utility.Vector3dVector(np_pcd)
+        pcd_filename = path.replace(".bin", ".pcd")
+        o3d.io.write_point_cloud(pcd_filename, pcd)
+        if delete_source:
+            os.remove(path)
+        return pcd_filename
+
+    def set_path(self, path):
+        self.path = path
+
+    def bin_operation(self, file_path, actual_path):
+        pcd_path = ValidateDimension.convert_bin_to_pcd(file_path)
+        self.converted_files.append(pcd_path)
+        return pcd_path.split(actual_path)[-1][1:]
+
+    def pcd_operation(self, file_path, actual_path):
+        if self.validate_pcd(file_path):
+            return file_path.split(actual_path)[-1][1:]
+        else:
+            return file_path
+
+    def validate(self):
+        """
+            Validate the directory structure for kitty and point cloud format.
+        """
+        if not self.path:
+            return
+        actual_path = self.path
+        for root, dirs, files in os.walk(actual_path):
```

Comment:
@zhiltsov-max , should the code (import velodyne dataset) migrate to datumaro? Or we prefer to keep it here for a while.